app/javascript/app/manage_users/email_form_input.js

```diff
@@ -0,0 +1,8 @@
+export const buildEmailAddressFormInput = (email) => {
+  const input = document.createElement("input");
+  input.type = "hidden";
+  input.name = "user_emails[]";
+  input.className = "user_email_input";
+  input.value = email;
+  return input;
```

Comment:
IMO this is starting to veer into "should we just use a JS templating library" territory. Combined with the `updateEmailAddressList` in the other file, I feel like this would make for a great mustache template. I know mustache doesn't support callbacks out of the box.... so maybe that's a case for a small React form or Vue.js (which I haven't used) seems to support event handlers as well.

Reply:
Yeah, I thought several times about pulling in React to handle this. It felt a little heavy handed. If you feel strongly I can go down that road though.

Reply:
You're the author, and work in this codebase more than I do at the moment, so I'll defer to you! Just saying that I think we're definitely approaching the point where it makes sense to switch to something soon.